Issues with AMD Plugin syntax

I was trying typescript@next (specifically v1.8.0-dev.20151120) and we extensively use AMD loader plugins, where we use `[module id]![resource id]`.  In v1.6.2 and earlier we would simply declare an ambient declaration describing our module, for example, we have a module that reverts to the CommonJS loader and we would do something like this to load a CommonJS only module in our `.d.ts` file:

```typescript
declare module 'intern/dojo/node!http' {
	import http = require('http');
	export = http;
}
```

And then we would import it in module:

```typescript
import * as http from 'intern/dojo/node!http';
```

And everything worked fine.  But it seems that at some point, `tsc` appears to only be matching up to the bang (`!`) and either reporting the module not found, or matching returning whatever ambient declaration it matches and even with the ambient declaration above, I am getting `error TS2307: Cannot find module 'intern/dojo/node'.`

I tried to find the related change, but only thing I was aware of was #2709 and that doesn't look like that has been addressed yet.

Is there anyway to get `tsc` to revert to the previous behaviour?

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -1034,10 +1034,6 @@ namespace ts {
                 return;
             }
 
-            if (moduleName.indexOf("!") >= 0) {
-                moduleName = moduleName.substr(0, moduleName.indexOf("!"));
-            }
-
             const isRelative = isExternalModuleNameRelative(moduleName);
             if (!isRelative) {
                 const symbol = getSymbol(globals, "\"" + moduleName + "\"", SymbolFlags.ValueModule);

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -1028,7 +1028,7 @@ namespace ts {
 
             // Module names are escaped in our symbol table.  However, string literal values aren't.
             // Escape the name in the "require(...)" clause to ensure we find the right symbol.
-            let moduleName = escapeIdentifier(moduleReferenceLiteral.text);
+            const moduleName = escapeIdentifier(moduleReferenceLiteral.text);
 
             if (moduleName === undefined) {
                 return;

--- a/tests/baselines/reference/bangInModuleName.js
+++ b/tests/baselines/reference/bangInModuleName.js
@@ -1,0 +1,23 @@
+//// [tests/cases/compiler/bangInModuleName.ts] ////
+
+//// [a.d.ts]
+
+
+declare module "http" {
+}
+
+declare module 'intern/dojo/node!http' {
+    import http = require('http');
+    export = http;
+}
+
+//// [a.ts]
+
+/// <reference path="a.d.ts"/>
+
+import * as http from 'intern/dojo/node!http';
+
+//// [a.js]
+/// <reference path="a.d.ts"/>
+define(["require", "exports"], function (require, exports) {
+});

--- a/tests/baselines/reference/bangInModuleName.symbols
+++ b/tests/baselines/reference/bangInModuleName.symbols
@@ -1,0 +1,21 @@
+=== tests/cases/compiler/a.ts ===
+
+/// <reference path="a.d.ts"/>
+
+import * as http from 'intern/dojo/node!http';
+>http : Symbol(http, Decl(a.ts, 3, 6))
+
+=== tests/cases/compiler/a.d.ts ===
+
+
+declare module "http" {
+}
+
+declare module 'intern/dojo/node!http' {
+    import http = require('http');
+>http : Symbol(http, Decl(a.d.ts, 5, 40))
+
+    export = http;
+>http : Symbol(http, Decl(a.d.ts, 5, 40))
+}
+

--- a/tests/baselines/reference/bangInModuleName.types
+++ b/tests/baselines/reference/bangInModuleName.types
@@ -1,0 +1,21 @@
+=== tests/cases/compiler/a.ts ===
+
+/// <reference path="a.d.ts"/>
+
+import * as http from 'intern/dojo/node!http';
+>http : typeof http
+
+=== tests/cases/compiler/a.d.ts ===
+
+
+declare module "http" {
+}
+
+declare module 'intern/dojo/node!http' {
+    import http = require('http');
+>http : typeof http
+
+    export = http;
+>http : typeof http
+}
+

--- a/tests/cases/compiler/bangInModuleName.ts
+++ b/tests/cases/compiler/bangInModuleName.ts
@@ -1,0 +1,17 @@
+// @module: amd
+
+// @filename: a.d.ts
+
+declare module "http" {
+}
+
+declare module 'intern/dojo/node!http' {
+    import http = require('http');
+    export = http;
+}
+
+// @filename: a.ts
+
+/// <reference path="a.d.ts"/>
+
+import * as http from 'intern/dojo/node!http';

